dashboard/src/actions/overviewActions.js

```diff
@@ -215,7 +215,7 @@ export const publishDataset =
       const savedRuns = getState().overview.savedRuns;
 
       const response = await API.post(
-        `${endpoints?.api?.datasets_update}/${dataset.resource_id}?access=${updateValue}`
```

Comment:
I liked `update`, although I suppose `modify` works as well. Why the change? I suppose you could argue that *neither* really clean encompasses "delete"; but I certainly don't think "modify" does any better.

We're gradually integrating this into an all-purpose "resource" API, which could also take over some evolution of `datasets_detail` as well, and might well take over the `datasets_metadata` `GET`/`PUT` APIs. That might argue for making this endpoint label simply `datasets`.

E.g.,
* `GET /datasets` would be the new `datasets_list`,
* `GET /datasets/{id}` would integrate `datasets_detail` and the `GET` on `datasets_metadata` with query parameters to specify a "view", `?metadata` to get metadata, etc.
* `PUT /datasets/{id}` updates METADATA via the JSON `{"metadata":{...}}` body (although it might be more "RESTful" to integrate this as a resource *action* under `POST` since we wouldn't be treating `metadata` as a separate sub-resource as `/datasets/metadata` implies)
* `POST /datasets/{id}` allows publish, "re-own" and maybe others, via query parameters ...
* `DELETE /datasets/{id}` allows deleting the resource.

Reply:
I, too, have a mild preference for `update` over `modify`, but I think I like the more resource-oriented API which Dave is proposing here.

I'm assuming that most of those changes should be dealt with in separate PRs...that _this_ PR will be limited to just `DELETE`.  But, it would be good if this PR laid the groundwork for those as follow-ons.

(However, I think I still prefer `metadata` as a sub-resource of `dataset/{id}`, but that's a conversation for a different time and place.)

Reply:
> (However, I think I still prefer `metadata` as a sub-resource of `dataset/{id}`, but that's a conversation for a different time and place.)

Yeah, I'm on the fence about that. Metadata is an aspect of a dataset, and treating it as direct attributes is fine; but thinking of metadata keys as a sub-resource doesn't really bother me, either... and that might argue for `/datasets/{id}/metadata`.

But to re-emphasize what we have both said here, *this* PR shouldn't go beyond integrating `DELETE`.